monkey/monkey_island/cc/dal/ISimulationRepository.py

```diff
@@ -0,0 +1,11 @@
+from abc import ABC
+
+
+class ISimulationRepository(ABC):
+    # TODO define simulation object. It should contain metadata about simulation,
+    # like start, end times, mode and last forced stop of all monkeys
```

Comment:
What's the intended use of this metadata? Do we have a good way of detecting simulation start and end if agents are started manually? It's doable, but it adds complexity. What's the value?

Reply:
This metadata is shown in the report to indicate the duration of the simulation. The way we do it right now is by the start of first monkey and by the death of the last one. I suppose we could query all of the telemetries, sort them by time and take the first wake-up and last "shutting down" telemetry, but it's just easier to set these parameters as the telemetries are incoming.